sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/BaseClientBuilder.java

```diff
@@ -118,6 +119,13 @@ private HttpPipelinePolicy makeValidationPolicy() {
         return builder.build();
     }
 
+    /**
+     * Adds an optional encryption policy that decrypts encrypted blobs.
+     * @param policies The list of policies to add an optional encryption policy to.
+     */
+    protected void addOptionalEncryptionPolicy(List<HttpPipelinePolicy> policies) {
```

Comment:
I think this should be added into each encryption client builder instead of the BaseClientBuilder as this should only be applicable to two builders.

Reply:
we call this in our base build pipeline method so the decryption policy can be added in the beginning of the pipeline. If I just added it in the two clients that need it then I would need to rewrite a lot of the base builder